templates/admin/user/edit.tmpl

```diff
@@ -94,45 +94,45 @@
 
 				<div class="inline field">
 					<div class="ui checkbox">
-						<label><strong>{{.locale.Tr "admin.users.is_activated"}}</strong></label>
-						<input name="active" type="checkbox" {{if .User.IsActive}}checked{{end}}>
+						<label for="active"><strong>{{.locale.Tr "admin.users.is_activated"}}</strong></label>
+						<input id="active" name="active" type="checkbox" {{if .User.IsActive}}checked{{end}}>
```

Comment:
active seems like a particularly bad name for an id.

Reply:
Single word is always bad. And there is a guideline (3) for code quality: https://docs.gitea.io/en-us/guidelines-frontend/#gitea-specific-guidelines

Reply:
> active seems like a particularly bad name for an id.

Can you explain why please? ID has been taken from `name`. I think it will be good to avoid collisions.

Reply:
Changed all occurrences to `"is_active"`.

Reply:
> I am thankful for the work provided regardless! heart

Thank you for your support :)


> Do you have an example of this regex? Or maybe I missed it.

~~~bash
# add id attr to an input and a for attr to their label located after
perl -0 -pe 's#(<input)((?:[^ >]| (?!id=))*name="([^"]+)"(?:[^ >]| (?!id=))*>[^<]+<label)((?:[^ >]| (?!for=))*>)#\1 id="\3"\2 for="\3"\4#msig'

# add id attr to an input and a for attr to their label located before
perl -0 -pe 's#(<label)((?:[^ >]| (?!for=))*>(?:[^<]|<(?!/label>))+</label>[^<]+<input)((?:[^ >]| (?!id=))*name="([^"]+)"(?:[^ >]| (?!id=))*>)#\1 for="\4"\2 id="\4"\3#msig'

# add for attr to a label located after its input which has id attr
perl -0 -pe 's#(<input[^>]*id="([^"]+)"[^>]*>[^<]+<label)((?:[^ >]| (?!for=))*>)#\1 for="\2"\3#msig'

# add for attr to a label located before its input which has id attr and its not a hidden
git grep -l -P '<div [^>]*class="(|[^"]+ )(ui|checkbox)( | [^"]* )(ui|checkbox)( [^" ]*|)">' templates/ | xargs -I% perl -i -0 -pe 's#(<label)((?:[^ >]| (?!for=))*>(?:[^<]|<(?!/label>))+</label>[^<]+<input(?:[^ >]| (?!type="hidden"))*id="([^"]+)"(?:[^ >]| (?!type="hidden"))*>)#\1 for="\3"\2#msig' %

# add id attr to an input from a for attr of their label located before
perl -0 -pe 's#(<label[^>]*for="([^"]+)"[^>]*>(?:[^<]|<(?!/label>))+</label>[^<]+<input)((?:[^ >]| (?!id=))*>)#\1 id="\2"\3#msig'

# add id attr to an input from a for attr of their label located after
perl -0 -pe 's#(<input)((?:[^ >]| (?!id=))*>[^<]+<label[^>]*for="([^"]+)"[^>]*>)#\1 id="\3"\2#msig'
~~~

The radio case is a special one because shares the css class `checkbox`, but is possible to add extra conditions to RE for generate an ID as a concatenation of `name`_`value` to avoid collisions. I forgot that in the initial RE, but the examples are a good starting point to refactor all that UI components easily.

The order of classes apparently is more stable than I suspected, so the RE is unnecessary more complex.

Reply:
> Single word is always bad. And there is a guideline (3) for code quality: https://docs.gitea.io/en-us/guidelines-frontend/#gitea-specific-guidelines

The guideline (3) defined ID format for *javascript uses*, which is not the case. For this changes applies guideline (2) only.